### ISSUE TITLE
Heap metrics are missing w/ native graalvm image

### ISSUE THREAD

**Issue by Haarolean** [micrometer-metrics/micrometer#4323]
**Describe the bug**
Heap metrics (e.g. `jvm_memory_used_bytes`) are missing while running micrometer via spring boot actuator within native graalvm image.

**Environment**

 - 1.11.5 (via spring-boot actuator starter)
 - prometheus
 - OS: macOS
 - Java version: 17

**To Reproduce**
How to reproduce the bug:

Run this demo: https://github.com/Haarolean/micrometer-graal-demo
Works as a jar, doesn't work as a native image (`mvn native:build` and run `./target/demo` after).

**Expected behavior**
It works

**Additional context**
Add any other context about the problem here, e.g. related issues.

**Comment by shakuzen** [comment]
> Java version: 17

What version of GraalVM are you using to compile to a native image? https://github.com/oracle/graal/issues/4510 added support for this in the June 2023 release, it looks like.

**Comment by Haarolean** [comment]
Sorry, I forgot the obvious! 😅

I'm actually using GraalVM 22.3.3 Java 17 CE. 
I've tried this with both compiling the demo myself and via CI (with `FROM ghcr.io/graalvm/native-image:ol8-java17-22` docker base image).

**Comment by shakuzen** [comment]
I was able to reproduce the issue on GraalVM 22.3.3, but that's expected I think. It's unfortunately unclear from the milestone attached to the issue I linked in which version exactly it should be fixed, but I did try with the latest (see below) and the issue is fixed there. I don't think there's anything we can do in Micrometer and so the answer is to use a more recent version of GraalVM - I believe GraalVM 23.0 or more recent.

```
openjdk version "21" 2023-09-19
OpenJDK Runtime Environment GraalVM CE 21+35.1 (build 21+35-jvmci-23.1-b15)
OpenJDK 64-Bit Server VM GraalVM CE 21+35.1 (build 21+35-jvmci-23.1-b15, mixed mode, sharing)
```
